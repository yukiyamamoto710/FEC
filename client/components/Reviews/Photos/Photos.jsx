import React from 'react';
import PropTypes from 'prop-types';

const Photos = (props) => {
  const { photos } = props;
  if (photos.length !== 0) {
    return (
      <div
        data-testid="photos"
        className="photoContainer"
      >
        {photos.map((i) => (
          <div
            data-testid="img"
            key={i.id + i}
          >
            <img
              alt=""
              src={i.url}
              className="photos"
            />
          </div>
        ))}
      </div>
    );
  }
  return <div data-testid="null" />;
};

Photos.propTypes = {
  photos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      url: PropTypes.string,
    }),
  ),
};

Photos.defaultProps = {
  photos: [{}],
};
export default Photos;
