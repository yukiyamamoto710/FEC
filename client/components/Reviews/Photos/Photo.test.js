/**
* @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Photos from './Photos';

afterEach(cleanup);

describe('render correct', () => {
  it('when photos.legnth is 0, should not show', () => {
    const test = [];
    const { getByTestId } = render(<Photos photos={test} />);
    expect(getByTestId('null')).toBeInTheDocument();
  });
  it('when photos.legnth > 0, should show', () => {
    const test = [
      { id: 1, url: 'a' },
      { id: 2, url: 'b' },
      { id: 3, url: 'c' }];
    const { getByTestId } = render(<Photos photos={test} />);
    expect(getByTestId('photos')).toBeInTheDocument();
  });
  it('when photos.legnth > 0, should img show', () => {
    const test = [
      { id: 1, url: 'a' },
      { id: 2, url: 'b' },
      { id: 3, url: 'c' }];
    const { getAllByTestId } = render(<Photos photos={test} />);
    expect(getAllByTestId('img')).toHaveLength(test.length);
  });
});

it('matches snapShot', () => {
  const test = [
    { id: 1, url: 'a' },
    { id: 2, url: 'b' },
    { id: 3, url: 'c' }];
  const tree = renderer.create(
    <Photos
      photos={test}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
