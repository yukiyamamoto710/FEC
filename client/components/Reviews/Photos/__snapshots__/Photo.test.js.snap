// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="photoContainer"
  data-testid="photos"
>
  <div
    data-testid="img"
  >
    <img
      alt=""
      className="photos"
      src="a"
    />
  </div>
  <div
    data-testid="img"
  >
    <img
      alt=""
      className="photos"
      src="b"
    />
  </div>
  <div
    data-testid="img"
  >
    <img
      alt=""
      className="photos"
      src="c"
    />
  </div>
</div>
`;
